File: Praesentia-YKM/32401.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        String s = scanner.next();

        int count = 0;

        for (int i = 0; i < n; i++) {
            for (int j = i + 2; j < n; j++) {
                if (s.charAt(i) != 'A' || s.charAt(j) != 'A') {
                    continue;
                }

                boolean valid = true;
                int nCount = 0;

                for (int k = i + 1; k < j; k++) {
                    if (s.charAt(k) == 'A') {
                        valid = false;
                        break;
                    }
                    if (s.charAt(k) == 'N') {
                        nCount++;
                    }
                }

                if (valid && nCount == 1) {
                    count++;
                }
            }
        }

        System.out.println(count);
    }
}

File: YoonbeomSo/32401.kt
import java.util.Scanner

fun main() {
    val sc = Scanner(System.`in`)
    val n = sc.nextInt()
    val s = sc.next()

    var count = 0

    for (i in 0 until n) {
        if (s[i] != 'A') continue
        for (j in i + 2 until n) {
            if (s[j] != 'A') continue
            val mid = s.substring(i + 1, j)
            if (!mid.contains('A') && mid.count { it == 'N' } == 1) {
                count++
            }
        }
    }

    println(count)
}
